keda/templates/22-metrics-deployment.yaml

```diff
@@ -118,7 +118,10 @@ spec:
       {{- if .Values.volumes.metricsApiServer.extraVolumes }}
       {{- toYaml .Values.volumes.metricsApiServer.extraVolumes | nindent 8 }}
       {{- end }}
-      hostNetwork: {{ .Values.metricsServer.useHostNetwork }}
+      {{- if .Values.metricsServer.useHostNetwork }}
+      hostNetwork: true
+      dsnPolicy: ClusterFirstWithHostNet
```

Comment:
Maybe good to make this configurable?

Reply:
```suggestion
      dnsPolicy: ClusterFirstWithHostNet
```

Reply:
there's not a whole lot of options for dnsPolicy, and kube docs do say

```
 For Pods running with hostNetwork, you should explicitly set its DNS policy "ClusterFirstWithHostNet"
```

I'll make it a parameter if that makes more sense.

Reply:
Let's make it a parameter indeed since there are multiple options when not using a host network: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

Reply:
I would also revert it so that we go back to `hostNetwork: {{ .Values.metricsServer.useHostNetwork }}` and we just introduce a DNS policy variable

Reply:
yep, updated.